File: code_review/work/Kuribayashi/PHASE2/step2/HotelPlan.php
<?php
// PHASE.2 step.2 HotelPlanクラスを作成する
class HotelPlan
{
    // 喫煙ルームオプション代
    const SMOKER_COST = 1000;
    // ホテルの部屋別ランク料金
    const HOTEL_ROOM_RANK_ADD_FEES = [
        'normal' => 0,
        'bronze' => 3000,
        'silver' => 5000,
        'gold' => 8000
    ];
    // 食事料金
    const MEAL_ADD_FEES = [
        'breakfast' => 500,
        'dinner' => 800
    ];

    public $price = 0;


    public function __construct($price)
    {
        $this->price = $price;
    }

    public function useSmokerRoom()
    {
        $this->price = $this->price + self::SMOKER_COST;
    }

    public function eatBrakefast()
    {
        $this->price = $this->price + self::MEAL_ADD_FEES['breakfast'];
    }
    public function selectHotelRank($rank)
    {
      $this->price = $this->price + self::HOTEL_ROOM_RANK_ADD_FEES[$rank];
        echo self::HOTEL_ROOM_RANK_ADD_FEES[$rank] . "\n";
    }
    public function eatDinner()
    {
        $this->price = $this->price + self::MEAL_ADD_FEES['dinner'];
    }

/*
        $this->price = $this->price + self::HOTEL_ROOM_RANK_ADD_FEES = [
            'normal' => 0,
            'bronze' => 3000,
            'silver' => 5000,
            'gold' => 8000
        ];
        */


}

File: code_review/question/PHASE4/omija/hotel-plan.rb
class HotelPlan
  SMOKER_COST = 1000
  HOTEL_ROOM_RANK_ADD_FEES = {
      "normal" => 0,
      "bronze" => 3000,
      "silver" =>  5000,
      "gold" => 8000,
  }
  MEAL_ADD_FEES = {
      "breakfast" => 500,
      "dinner" => 800,
  }

  def initialize(price)
    @hotel_price = price
  end

  # PHASE.2 step.1 喫煙可能部屋を使うメソッドを追加する
  def is_smoker
    @hotel_price += SMOKER_COST
  end

  # PHASE.2 step.1 ホテルのランクを選択するメソッドを追加する
  def select_hotel_rank(hotel_room_rank)
    if HOTEL_ROOM_RANK_ADD_FEES.fetch(hotel_room_rank, nil).nil?
      p "正しいランクを選択してください"
      exit(0) 
    else
      @hotel_price += HOTEL_ROOM_RANK_ADD_FEES[hotel_room_rank]
    end
  end

  # PHASE.2 step.1 朝食を食べるメソッドを追加する
  def has_breakfast
    @hotel_price += MEAL_ADD_FEES["breakfast"]
  end

  # PHASE.2 step.1 夕食を食べるメソッドを追加する
  def has_dinner
    @hotel_price += MEAL_ADD_FEES["dinner"]
  end
 
  # 戻り値用メソッド
  def get_hotel_price
    @hotel_price
  end
end
